File: academic-system/src/main/java/org/example/model/Avaliacao.java
package org.example.model;
import org.example.exception.ExcecaoSistemaAcademico;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.Getter;

@Getter
public abstract class Avaliacao {
	
	@NotBlank(message = "O nome da avaliação não pode ser vazio.")
	private String nome;
	
	@PositiveOrZero(message = "Dados inválidos: valor não pode ser negativo.")
	private double valor;
	
	@PositiveOrZero(message = "Dados inválidos: peso não pode ser negativo.")
	private double peso;

	public Avaliacao(String nome, double valor, double peso) {
		this.nome = nome;
		this.valor = valor;
		this.peso = peso;
	}
}

File: academic-system-csharp/SistemaAcademico/src/SistemaAcademico/Models/Avaliacao.cs
using System.ComponentModel.DataAnnotations;

namespace SistemaAcademico.Models;

// Equivalente à classe abstrata Java: Avaliacao
// Lombok @Getter → propriedades C# com get; privado
// Jakarta @NotBlank → [Required], [MinLength(1)]
// Jakarta @PositiveOrZero → [Range(0, double.MaxValue)]
public abstract class Avaliacao
{
    [Required(ErrorMessage = "O nome da avaliação não pode ser vazio.")]
    [MinLength(1, ErrorMessage = "O nome da avaliação não pode ser vazio.")]
    public string Nome { get; private set; }

    [Range(0, double.MaxValue, ErrorMessage = "Dados inválidos: valor não pode ser negativo.")]
    public double Valor { get; private set; }

    [Range(0, double.MaxValue, ErrorMessage = "Dados inválidos: peso não pode ser negativo.")]
    public double Peso { get; private set; }

    protected Avaliacao(string nome, double valor, double peso)
    {
        Nome = nome;
        Valor = valor;
        Peso = peso;
    }
}
